process/asyncExecution/executionManager/executionManager_test.go

```diff
@@ -143,24 +144,355 @@ func TestExecutionManager_SetHeadersExecutor(t *testing.T) {
+				if bytes.Equal(blockHash, []byte("lastNotarizedExecResultHash")) {
+					counter += 1
+				}
+			},
+			SetLastExecutionResultCalled: func(result data.BaseExecutionResultHandler) {
+				if bytes.Equal(result.GetHeaderHash(), []byte("lastNotarizedExecResultHash")) {
+					counter += 1
+				}
+			},
+		}
+
+		args.ExecutionResultsTracker = &processMocks.ExecutionTrackerStub{
+			GetPendingExecutionResultsCalled: func() ([]data.BaseExecutionResultHandler, error) {
+				return []data.BaseExecutionResultHandler{}, nil
+			},
+			GetLastNotarizedExecutionResultCalled: func() (data.BaseExecutionResultHandler, error) {
+				return &block.BaseExecutionResult{
+					RootHash:   []byte("rootHash"),
+					HeaderHash: []byte("lastNotarizedExecResultHash"),
+				}, nil
+			},
+		}
+
+		args.Headers = &pool.HeadersPoolStub{
+			GetHeaderByHashCalled: func(hash []byte) (data.HeaderHandler, error) {
+				return nil, errExpected
+			},
+		}
+
+		em, _ := executionManager.NewExecutionManager(args)
+
+		pair := queue.HeaderBodyPair{
+			Header: &block.Header{
+				Nonce:    9,
+				PrevHash: []byte("lastNotarizedExecResultHash"),
+			},
+			Body: &block.Body{},
+		}
+
+		err := em.AddPairForExecution(pair)
+		require.Equal(t, errExpected, err)
+		require.Equal(t, 0, counter)
+	})
+
+	t.Run("if removing from pending execution results fails, the error should be propagated", func(t *testing.T) {
+		t.Parallel()
+
+		counter := 0
+		args := createMockArgs()
+		args.BlockChain = &testscommon.ChainHandlerStub{
+			GetLastExecutedBlockHeaderCalled: func() data.HeaderHandler {
+				return &block.HeaderV3{
+					Nonce: 10,
+				}
+			},
+			SetLastExecutedBlockHeaderAndRootHashCalled: func(header data.HeaderHandler, blockHash []byte, rootHash []byte) {
+				if bytes.Equal(blockHash, []byte("lastNotarizedExecResultHash")) {
+					counter += 1
+				}
+			},
+			SetLastExecutionResultCalled: func(result data.BaseExecutionResultHandler) {
+				if bytes.Equal(result.GetHeaderHash(), []byte("lastNotarizedExecResultHash")) {
+					counter += 1
+				}
+			},
+		}
+
+		args.ExecutionResultsTracker = &processMocks.ExecutionTrackerStub{
+			GetPendingExecutionResultsCalled: func() ([]data.BaseExecutionResultHandler, error) {
+				return []data.BaseExecutionResultHandler{}, nil
+			},
+			GetLastNotarizedExecutionResultCalled: func() (data.BaseExecutionResultHandler, error) {
+				return &block.BaseExecutionResult{
+					RootHash:   []byte("rootHash"),
+					HeaderHash: []byte("lastNotarizedExecResultHash"),
+				}, nil
+			},
+			RemoveFromNonceCalled: func(nonce uint64) error {
+				return errExpected
+			},
+		}
+
+		em, _ := executionManager.NewExecutionManager(args)
+
+		pair := queue.HeaderBodyPair{
+			Header: &block.Header{
+				Nonce:    9,
+				PrevHash: []byte("lastNotarizedExecResultHash"),
+			},
+			Body: &block.Body{},
+		}
+
+		err := em.AddPairForExecution(pair)
+		require.Equal(t, errExpected, err)
+		require.Equal(t, 2, counter)
```

Comment:
[nitpicking]

Checking this counter instead of actually checking that the expected functions have been called could be misleading.
You could have for example here `SetLastExecutionResultCalled` being called twice instead of calling `SetLastExecutedBlockHeaderAndRootHashCalled` + `SetLastExecutionResultCalled`